### PR TITLE
NORALLY: fixed entity filter ordering based on the filter dependencies

### PR DESCRIPTION
**Pull request by uppoju**
## Description  
Filter ordering is changed based on the filter dependencies.

## Checklist
- [x] Pull Request Created
- [ ] Unit tests have been added
- [ ] Integration/Functional test cases have been written and automated

### CODE REVIEW

File: gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/EntityFilterRegistry.java [hunk 1]
```diff
@@ -8,22 +8,44 @@
...
+import java.util.*;
 
-import static java.util.Collections.unmodifiableSet;
+import static java.util.Collections.unmodifiableCollection;
 
 @Singleton
 public class EntityFilterRegistry {
 
     private final Collection<EntityFilter> entityFilters;
+    private Map<String, EntityFilter> sortedFilters = new LinkedHashMap<>();
```
Comment (nikhilahuja) [inline gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/EntityFilterRegistry.java:19]: This is temporary so we can avoid it as a member. Pass it as an argument to the method.

File: gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/entityfilters/StoredPasswordFilter.java [hunk 2]
```diff
@@ -21,7 +21,8 @@
             PolicyFilter.class,
             ServiceFilter.class,
             JDBCConnectionFilter.class,
-            CassandraConnectionFilter.class).collect(Collectors.toSet());
+            CassandraConnectionFilter.class,
+            SsgActiveConnectorFilter.class).collect(Collectors.toSet());
```
Comment (nikhilahuja) [inline gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/entityfilters/StoredPasswordFilter.java:25]: Can you also add `UnsupportedEntityFilter` and `GenericEntityFilter` to this list?

Comment (uppoju) [reply gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/entityfilters/StoredPasswordFilter.java:25]: ok

File: gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/EntityFilterRegistry.java (hunk 1 above)
Comment (uppoju) [reply gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/EntityFilterRegistry.java]: done

**Review by nikhilahuja — APPROVED** [review]

**Review by RaghuReddy — APPROVED** [review]
looks good

**Pull request merged**